lib/charms/traefik_route_k8s/v0/traefik_route.py

```diff
@@ -189,30 +200,43 @@ def _update_stored_external_host(self) -> None:
         allows for re-use both when the property is called and if the relation changes, so a
         leader change where the new leader checks the property will do the right thing.
         """
-        if self._charm.unit.is_leader():
-            for relation in self._charm.model.relations[self._relation_name]:
-                if not relation.app:
-                    self._stored.external_host = ""
-                    return
-                external_host = relation.data[relation.app].get("external_host", "")
-                self._stored.external_host = external_host or self._stored.external_host  # type: ignore
+        if not self._charm.unit.is_leader():
+            return
+
+        for relation in self._charm.model.relations[self._relation_name]:
+            if not relation.app:
+                self._stored.external_host = ""
+                self._stored.scheme = ""
+                return
+            external_host = relation.data[relation.app].get("external_host", "")
+            self._stored.external_host = external_host or self._stored.external_host  # type: ignore
+            scheme = relation.data[relation.app].get("scheme", "")
```

Comment:
Shouldn't we be explicit about the scheme here?

```suggestion
            scheme = relation.data[relation.app].get("scheme", "http")
```

Reply:
I think you're right, but I was following the existing pattern for `external_host`.
There are more odd things here that could use a follow-up PR:
- `_stored` gets overwritten every loop iteration.
- Why the provider needs to extract this from relation data? Seems like this was cloned from the requirer. Not sure.

Reply:
https://github.com/canonical/traefik-route-k8s-operator/issues/42